src/match_regex.c

```diff
@@ -184,8 +187,78 @@ static int mr_config_add_regex (mr_regex_t **re_head, /* {{{ */
 	}
 
 	return (0);
+} /* }}} int mr_add_regex */
+
+static int mr_config_add_regex (mr_regex_t **re_head, /* {{{ */
+		oconfig_item_t *ci)
+{
+	if ((ci->values_num != 1) || (ci->values[0].type != OCONFIG_TYPE_STRING))
+	{
+		log_warn ("`%s' needs exactly one string argument.", ci->key);
+		return (-1);
+	}
+
+	return mr_add_regex (re_head, ci->values[0].value.string, ci->key);
 } /* }}} int mr_config_add_regex */
 
+static int mr_config_add_meta_regex (llist_t **meta, /* {{{ */
+		oconfig_item_t *ci)
+{
+	char *key;
+	llentry_t *entry;
+	mr_regex_t *re_head;
+	int status;
+	char buffer[1024];
+
+	if ((ci->values_num != 2)
+		|| (ci->values[0].type != OCONFIG_TYPE_STRING)
+		|| (ci->values[1].type != OCONFIG_TYPE_STRING))
+	{
+		log_warn ("`%s' needs exactly two string arguments.", ci->key);
+		return (-1);
+	}
+
+	if (*meta == NULL)
+	{
+		*meta = llist_create();
+		if (*meta == NULL)
+		{
+			log_err ("mr_config_add_meta_regex: llist_create failed.");
+			return (-1);
+		}
+	}
+
+	key = ci->values[0].value.string;
+	entry = llist_search (*meta, key);
+	if (entry == NULL)
+	{
+		key = strdup (key);
+		if (key == NULL)
+		{
+			log_err ("mr_config_add_meta_regex: strdup failed.");
+			return (-1);
+		}
+		entry = llentry_create (key, NULL);
+		if (entry == NULL)
+		{
+			log_err ("mr_config_add_meta_regex: llentry_create failed.");
+			free (key);
+			return (-1);
+		}
+		/* key and entry will now be freed by mr_free_match(). */
+		llist_append (*meta, entry);
+	}
+
+	snprintf (buffer, sizeof (buffer), "%s `%s'", ci->key, key);
```

Comment:
At your option: I find `ci->key` and `key` a bit hard to read. How about renaming `key` to `meta_name`?